hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/fs/SpaceUsageCheckFactory.java

```diff
@@ -68,14 +69,16 @@ default SpaceUsageCheckFactory setConfiguration(Configuration conf) {
    * Defaults to {@link DUFactory} if no class is configured or it cannot be
    * instantiated.
    */
-  static SpaceUsageCheckFactory create(Configuration config) {
+  static SpaceUsageCheckFactory create(ConfigurationSource config) {
     Conf conf = OzoneConfiguration.of(config).getObject(Conf.class);
     Class<? extends SpaceUsageCheckFactory> aClass = null;
     String className = conf.getClassName();
     if (className != null && !className.isEmpty()) {
       try {
-        aClass = config.getClassByName(className)
-            .asSubclass(SpaceUsageCheckFactory.class);
+        aClass =
+            (Class<? extends SpaceUsageCheckFactory>)
```

Comment:
Can we keep `.asSubclass(SpaceUsageCheckFactory.class)` instead of the unchecked cast?